programs/github

```diff
@@ -21,14 +21,47 @@ function progressbar(f)
 	return left .. right
 end
 
-if action == 'clone' then
-	-- parse input
-	local repo, treeName, dest = select(2, ...)
+-- get flag arguments
+local acceptedFlags = {
+	['-b'] = 'branch',
+	['--branch'] = 'branch',
+	['-t'] = 'tag',
+	['--tag'] = 'tag'
+}
 
-	if not dest then
-		dest = treeName
-		treeName = 'master'
+local function findFlag(arg)
+	if arg.sub(1,1) == '-' then
+		if not acceptedFlags[arg] then
+			return printError(('Unknown flag argument %s'):format(arg))
```

Comment:
Let me look into the `printError()` function I assumed it would terminate
the program.

On Sat, Aug 27, 2016, 21:00 Eric Wieser <notifications@github.com> wrote:

> In programs/github
> <https://github.com/eric-wieser/computercraft-github/pull/11#discussion_r76524809>
> :
>
> >
> > -	if not dest then
> > -		dest = treeName
> > -		treeName = 'master'
> > +local function findFlag(arg)
> > +	if arg.sub(1,1) == '-' then
> > +		if not acceptedFlags[arg] then
> > +			return printError(('Unknown flag argument %s'):format(arg))
>
> Right, but it only exits this function. The script will continue and run,
> won't it, treating this as the dest/repo?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/eric-wieser/computercraft-github/pull/11/files/78509095ce2a421a6e2f7dd6d245dcc050539025#r76524809>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA41FoXaVaR7Wmk1f6zugjhbdAUseYZZks5qkOvOgaJpZM4JsvBA>
> .
>